src/cli.rs

```diff
@@ -43,28 +43,49 @@ pub struct Rover {
 
     #[structopt(skip)]
     #[serde(skip_serializing)]
-    pub env_store: RoverEnv,
+    pub env_store: RoverEnvWrapper,
+}
+
+#[derive(Debug)]
+pub struct RoverEnvWrapper {
+    pub inner: RoverEnv,
+}
+
+impl Default for RoverEnvWrapper {
+    fn default() -> Self {
+        if cfg!(test) {
+            RoverEnvWrapper {
+                inner: RoverEnv::mock(),
+            }
+        } else {
+            RoverEnvWrapper {
+                inner: RoverEnv::new(),
+            }
+        }
+    }
```

Comment:
i needed to break this out like this since `cfg!(test)` does not work in workspace crates :(

i don't love this pattern since it means we need to call `.inner` after each `.env_store`, but... unfortunately Rust's orphan rules make it so we can't implement `Default` on an external type. if anybody has any ideas here i'd love to here them.

satisfied requirements:

1) `RoverEnvKey` can be imported across subcrates (no circular dependencies!)
2) `Rover::env_store` has a `Default` implementation that takes advantage of `cfg!(test)`

unsatisfied nice to have:

1) be able to call `rover_env.get()` like we used to instead of `rover_env.inner.get()`

Reply:
nevermind